tests/ui/float_equality_without_abs.stderr

```diff
@@ -0,0 +1,70 @@
+error: float equality check without `.abs()`
+  --> $DIR/float_equality_without_abs.rs:4:5
+   |
+LL |     (a - b) < f32::EPSILON
+   |     ^^^^^^^^^^^^^^^^^^^^^^ help: add `.abs()`: `(a - b).abs()`
+   |
+   = note: `-D clippy::float-equality-without-abs` implied by `-D warnings`
+
+error: float equality check without `.abs()`
+  --> $DIR/float_equality_without_abs.rs:13:13
+   |
+LL |     let _ = (a - b) < f32::EPSILON;
+   |             ^^^^^^^^^^^^^^^^^^^^^^ help: add `.abs()`: `(a - b).abs()`
```

Comment:
This suggestion doesn't work. If applied, this will transform the code like this:

```rust
let _ = (a - b) < f32::EPSILON; // before
let _ = (a - b).abs(); // after
```

A better way to do this would be to use `span_lint_and_then` and then use the `db.span_suggestion` to only add the above suggestion to the span of `(a - b)`.

Reply:
Oh you're right! Thank you @flip1995 and sorry that I didn't see that earlier! I've corrected it on my own fork. Should I open a new PR for this fix? I've also started to use `utils::Sugg`. But I had some issues while parsing the expression, the simple way of 
```rust
let suggestion = format!("{}.abs()", sugg::Sugg::hir(cx, &lhs, "..").maybe_par());
```
always brought up doubled parentheses. E.g. `(a - b)` -> `((a - b))`. That's why I took a longer way and assembled the suggestion by myself. Maybe you could have a look on the code, it's all in this commit [here](https://github.com/1c3t3a/rust-clippy/commit/2d853148d72d49956052cf1cdb5f3be18c85a9fc).

Reply:
The linked commit looks good to me.

Yes please open a new PR about this!